### PR TITLE
Alert on change for Dropdown and MultipleSelect

### PR DESCRIPTION
**Pull request by bjosttveit**
## Description

Adds `alertOnChange` property to `Dropdown` and `MultipleSelect`. These work exactly the same way as RadioButtons and Checkboxes respectively, meaning that it will show an alert when you "remove a value". For Radiobuttons and Dropdown this means that no alert is shown if the value was empty before, but does show when the previous value was not null. For checkboxes and multiple select it shows an alert when you uncheck or remove a selected value, this includes when you click the multiple select's delete all button.

The design system was upgraded for this to work, and the `LegacySelect` has been upgraded to the new `Combobox` component from the design system for `Dropdown`, `MutlipleSelect`, and `FileUploadWithTag` (and everywhere else).

Due to design system changes in this version I had to update some other components as well, most notably `Paragraph`. The design system `Paragraph` no longer lets you choose the tag to use, so I had to conditionally wrap it instead. When making these changes I noticed that our language function not only replaces the so called "root element" with a span (to support inline content in buttons etc.), it also replaces it even if there is no "root" but rather multiple block level elements, causing cases where I personally think the `p`-element should have been left in place to also be replaced with a span. I made a small change to this logic to not only check if it is at the top-level, but also check if it has any siblings. That way, if you have a say paragraph and a list as the content, the p-element will be left in place. I have not noticed this messing up the texts anywhere else, but I can remove it in case it does.

Another thing related to the Paragraph element, some of the unit tests did not actually do anything. This is because it did a `.forEach` with an async function which would finish immediately and no expect-calls would therefore happen before the function returned, which just meant they were always green. This has been fixed by changing it to a `for..of` loop instead of a `.forEach`.

## Related Issue(s)

- closes https://github.com/Altinn/app-frontend-react/issues/1233

## Verification/QA

- Manual functionality testing
  - [x] I have tested these changes manually
  - [ ] Creator of the original issue (or service owner) has been contacted for manual testing (or will be contacted when released in alpha)
  - [ ] No testing done/necessary
- Automated tests
  - [x] Unit test(s) have been added/updated
  - [x] Cypress E2E test(s) have been added/updated
  - [ ] No automatic tests are needed here (no functional changes/additions)
  - [ ] I want someone to help me make some tests
- UU/WCAG ([follow these guidelines](https://aksel.nav.no/god-praksis/artikler/utvikling?tema=universell-utforming) until we have our own)
  - [ ] I have tested with a screen reader/keyboard navigation/automated wcag validator
  - [ ] No testing done/necessary (no DOM/visual changes)
  - [ ] I want someone to help me perform accessibility testing
- User documentation @ [altinn-studio-docs](https://github.com/Altinn/altinn-studio-docs)
  - [ ] Has been added/updated
  
  - [ ] No functionality has been changed/added, so no documentation is needed
  - [ ] I will do that later/have created an issue
  
- Support in Altinn Studio
  - [ ] Issue(s) created for support in Studio
  
  - [ ] This change/feature does not require any changes to Altinn Studio
- Sprint board
  - [x] The original issue (or this PR itself) has been added to the Team Apps project and to the current sprint board
  - [ ] I don't have permissions to do that, please help me out
- Labels
  - [x] I have added a `kind/*` label to this PR for proper release notes grouping
  - [ ] I don't have permissions to add labels, please help me out

### CODE REVIEW

File: src/index.css [hunk 1]
```diff
@@ -11,6 +11,11 @@
   color: var(--semantic-text-neutral-default);
 }
 
+/* Workaround for the design-system aggressively setting the font to 'Inter' */
+[class^='fds-'] {
+  font-family: var(--font-family);
+}
```
Comment (olemartinorg) [inline src/index.css:17]: :sweat_smile: :clap:

File: test/e2e/integration/frontend-test/all-process-steps.ts [hunk 2]
```diff
@@ -70,10 +70,8 @@ function testConfirmationPage() {
 
   cy.snapshot('confirm');
 
-  // TODO: Enable this when it doesn't break custom receipt. It would be nice to make sure refreshing on a confirm step
-  // restores all state and we're not stuck in any way here.
-  // cy.reloadAndWait();
-  // cy.get(appFrontend.confirm.container).should('be.visible');
+  cy.reloadAndWait();
+  cy.get(appFrontend.confirm.container).should('be.visible');
```
Comment (olemartinorg) [inline test/e2e/integration/frontend-test/all-process-steps.ts:74]: Oh, nice! :raised_hands: That should probably have been commented back in a good while ago.

**Review by olemartinorg — APPROVED** [review]
Great stuff! :godmode:

File: test/e2e/integration/frontend-test/all-process-steps.ts (hunk 2 above)
Comment (bjosttveit) [reply test/e2e/integration/frontend-test/all-process-steps.ts:74]: yeah probably for v4.0.0 😅

**Review by adamhaeger — APPROVED** [review]

**Pull request merged**